src/Nethermind/Nethermind.Evm/EvmPooledMemory.cs

```diff
@@ -24,21 +30,34 @@ public class EvmPooledMemory : IEvmMemory
 
         public void SaveWord(in UInt256 location, Span<byte> word)
         {
-            CheckMemoryAccessViolation(in location, WordSize);
-            UpdateSize(in location, WordSize);
+            CheckMemoryAccessViolation(in location, in WordSize256);
+            UpdateSize(in location, in WordSize256);
 
-            if (word.Length < WordSize)
+            int offset = (int)location;
+            if (word.Length == WordSize)
             {
-                Array.Clear(_memory!, (int)location, WordSize - word.Length);
+                // Direct 256bit register copy rather than invoke Memmove
+                Unsafe.WriteUnaligned(
+                    ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(_memory), offset),
+                    Unsafe.As<byte, Vector256<byte>>(ref MemoryMarshal.GetReference(word))
+                );
+            }
+            else if (word.Length < WordSize)
+            {
+                Array.Clear(_memory!, offset, WordSize - word.Length);
+                word.CopyTo(_memory.AsSpan(offset + WordSize - word.Length, word.Length));
+            }
+            else
+            {
+                // Should never be possible, but just for completeness of if states
+                ThrowOutOfGasException();
             }
-
-            word.CopyTo(_memory.AsSpan((int)location + WordSize - word.Length, word.Length));
         }
 
         public void SaveByte(in UInt256 location, byte value)
         {
-            CheckMemoryAccessViolation(in location, WordSize);
-            UpdateSize(in location, 1);
+            CheckMemoryAccessViolation(in location, in WordSize256);
+            UpdateSize(in location, in UInt256.One);
```

Comment:
Minor: both of those calculate `position + length` and are almost always invoked together, maybe we should cover that too?

Reply:
Done